pkg/config/output.go

```diff
@@ -84,7 +86,12 @@ func (p *PipelineConfig) updateEncodedOutputs(req egress.EncodedOutput) error {
 		return errors.ErrInvalidInput("multiple stream outputs")
 	}
 	if stream != nil {
-		conf, err := p.getStreamConfig(types.OutputTypeRTMP, stream.Urls)
+		outputType := types.OutputTypeRTMP
+		if len(stream.Urls) > 0 && strings.HasPrefix(stream.Urls[0], "srt://") {
```

Comment:
I may be cleaner to have a `switch` statement on the prefix rather than defaulting to rtmp.

We should also validate that all the streams use the same protocol as the code currently doesn't allow mixing SRT and RTMP.

Reply:
Changed to switch statement. Not sure how to validate that the streams use the same protocol. Would be nice if there were some unit tests to catch this edge case.

Reply:
Up to this change, all streams were always RTMP, so there was no reason to check that they were all the same protocol. It is needed now that we support more than 1 protocol. Adding a unit test for this is a good idea.